tests/datasource/fluent/test_redshift_datasource.py

```diff
@@ -0,0 +1,136 @@
+import logging
+
+import pytest
+from pytest_mock import MockerFixture
+
+from great_expectations.data_context import EphemeralDataContext
+from great_expectations.datasource.fluent.redshift_datasource import (
+    RedshiftConnectionDetails,
+    RedshiftDsn,
+    RedshiftSSLModes,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def build_connection_string(scheme, username, password, host, port, database, ssl_mode):
+    return f"{scheme}://{username}:{password}@{host}:{port}/{database}?sslmode={ssl_mode}"
+
+
+@pytest.fixture
+def scheme():
+    return "redshift+psycopg2"
+
+
+@pytest.mark.unit
+@pytest.mark.redshift
+def test_create_engine_is_called_with_expected_kwargs_using_connection_string(
+    sa,
+    mocker: MockerFixture,
+    ephemeral_context_with_defaults: EphemeralDataContext,
+    scheme,
+):
+    create_engine_spy = mocker.patch.object(sa, "create_engine")
+
+    username = "username"
+    password = "password"
+    host = "host"
+    port = 1234
+    database = "database"
+    ssl_mode = "allow"
+    connection_string = build_connection_string(
+        scheme, username, password, host, port, database, ssl_mode
+    )
+
+    expected_kwargs = RedshiftDsn(
+        connection_string,
```

Comment:
Why do we need to use both the connection_string and all the other arguments when instantiating the `RedshiftDsn`? If we just specify the connection string, will the others be inferred? Maybe we also have to pass scheme (I seem to recall that is necessary)?